pkg/server/apiextensions.go

```diff
@@ -394,12 +387,12 @@ func (c *apiBindingAwareCRDLister) get(clusterName logicalcluster.Name, name str
 	for _, obj := range objs {
 		apiBinding := obj.(*apisv1alpha1.APIBinding)
 
-		if !conditions.IsTrue(apiBinding, apisv1alpha1.InitialBindingCompleted) {
-			continue
-		}
-
 		for _, boundResource := range apiBinding.Status.BoundResources {
-			if boundResource.Group == group && boundResource.Resource == resource {
+			// identity is empty string if the request is coming from a regular workspace client.
```

Comment:
IMHO using the presence or absence of stringified data to drive logic is fragile - I just spent a long time reworking assumptions like this in keyfuncs to be more concrete & use the type system to be clearer. Do we *need* this implicit assumption or is there perhaps some other factoring / metadata / etc that could help us make the distinction explicitly?

Reply:
also referring to @sttts 🙏

Reply:
we already have this today basically everywhere including in this file, i.e. identity empty vs. identity non-empty. This code here just matches that mattern.

What I agree with is that the comment should not distinguish between clients here, but talk about the request: with-identity and without-identity requests.